open_spiel/games/maedn.h

```diff
@@ -0,0 +1,281 @@
+  // Some values are not part of ObservationTensor (like prev_player_ and 
+  // prev_dice_) and so have to be given from outside. History is not part
+  // of ObservationTensor either, so calls to UndoAction will cause undefined
+  // behaviour!
+  void FromObservationTensor(Player player, 
+                             absl::Span<float> values, 
+                             Player prev_player,
+                             int prev_dice);
+
+  int dice() const { return dice_; }
+
+ protected:
+  void DoApplyAction(Action move_id) override;
+
+ private:
+  void SetupInitialBoard();
+  void RollDice(int outcome);
+  std::pair<int, int> GetFieldsFromAction(Action action, 
+                                          Player player, 
+                                          int dice) const;
+  int RelPosToAbsPos(int relative_position, int position) const;
+  int AbsPosToRelPos(int absolute_position, int position) const;
+  int GetPlayersFirstField(Player player) const;
+  
+  int PlayerToPosition(Player player) const {
+    // Position is equal to player except if two players play on opposite
+    // sides, in this case position of player 1 is 2. For completeness,
+    // in this case position of player 2 is 1, so that even for iterations
+    // over 4 players no position is used twice.
+    return num_players_ == 2 && two_players_opposite_ && 
+           (player == 1 || player == 2) ?
+           3 - player : player;
+  }
+
+  bool AllInGoal(Player player) const;
+  Player cur_player_;
+  Player prev_player_;
+  const bool two_players_opposite_;
+  int turns_;
+  int dice_;             // Current dice roll.
+  int prev_dice_;        // Last dice roll.
+  std::vector<int> out_; // Number of pieces of each player outside of field.
+
+  // Board consists of 40 common fields, starting with the set-in field of 
+  // player 0. After that, four goal fields of each player follow, beginning
+  // with player 0 again.
+  // Player 0 starts on field 0, goes up to field 39 and continues into
+  // goal fields 40-43.
+  // Player 1 starts on field 10, goes up to field 39, continues from 0 to 9
+  // and jumps from 9 to 44-47.
+  // Player 2 starts on field 20, goes up to field 39, continues from 0 to 19
+  // and jumps from 19 to 48-51.
+  // Player 3 starts on field 30, goes up to field 39, continues from 0 to 29
+  // and jumps from 29 to 52-55.
+  std::vector<int> board_;
+  std::vector<TurnHistoryInfo> turn_history_info_;  // Info needed for Undo.
+};
+
+class MaednGame : public Game {
+ public:
+  explicit MaednGame(const GameParameters& params);
+
+  int NumDistinctActions() const override { return kNumDistinctActions; }
+
+  std::unique_ptr<State> NewInitialState() const override {
+    return std::unique_ptr<State>(new MaednState(
+        shared_from_this(), two_player_opposite_));
+  }
+
+  // Classic six sided dice.
+  int MaxChanceOutcomes() const override { return 6; }
+
+  // There is arbitrarily chosen number to ensure the game is finite.
```

Comment:
nit: reads better without the "there is", just: "Arbitrarily chosen number to ensure the game is finite."

Reply:
Fixed.